### ISSUE TITLE
No matching manifest for linux/arm/v7

### ISSUE THREAD

**Issue by felipejoribeiro** [shellhub-io/shellhub#2440]
### Description

Last week, the provided command for installing agents ceased to work:
```
curl -sSf "http://YYYY/install.sh?tenant_id=XXXX" | sh
```

We are trying to configure shellhub on our raspberry pies. We have done that in the past months without problems, but the following error started appearing preventing us from registering new agents:

```
Unable to find image 'shellhubio/agent:v0.9.4' locally
v0.9.4: Pulling from shellhubio/agent
docker: no matching manifest for linux/arm/v7 in the manifest list entries.
See 'docker run --help'.
```

Our raspberry pi image is the following:

``` 
pi@rasp:~ $ cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
```

### Edition

Community

### Version

0.9.4

**Comment by gustavosbarreto** [comment]
Recently Docker made some changes related to manifest handling. Not sure if this affected old versions of docker.

I did some tests to make sure the `linux/arm/v7` manifest exists.

```
$ docker manifest inspect shellhubio/agent:v0.9.4
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1784,
         "digest": "sha256:3e5d6f97864d9377a97ac7569afea50de1e1aec49b77056c7ef8f93a1700125e",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1784,
         "digest": "sha256:884f40b10b46ef27d3c9644dac9c43a08d2e4f575b5430aeb165848e76f67441",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v6l"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1784,
         "digest": "sha256:de8b293cb077781a8e1370ba0392e8b05478543084cfb0d5b956a531c67fab4d",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v7l"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1784,
         "digest": "sha256:0e029f36eadb51736ef3b18d9e7c771b2e667ab1396f66477eb2b237613873ec",
         "platform": {
            "architecture": "arm64",
            "os": "linux",
            "variant": "v8"
         }
      }
   ]
}
```

```
$ docker pull --platform linux/arm/v7 shellhubio/agent:v0.9.4
v0.9.4: Pulling from shellhubio/agent
ac4f08a50f8c: Pull complete
ccfadbfba020: Pull complete
6b874748d339: Pull complete
58a82240d886: Pull complete
de57960c1451: Pull complete
1bb1cfe71b89: Pull complete
918c55bc5003: Pull complete
Digest: sha256:2e010e00897a9fd4a520a73b45031f88e16739935417d4c7471e8a15c8eb5042
Status: Downloaded newer image for shellhubio/agent:v0.9.4
```

@felipejoribeiro What is the Docker version on your rasp?

**Comment by felipejoribeiro** [comment]
Thank you for the prompt response:
```
pi@rasp:~ $ docker --version
Docker version 23.0.1, build a5ee5b1
```

**Comment by gustavosbarreto** [comment]
It makes no sense. The image exists in the docker registry:

![Screenshot (26)](https://user-images.githubusercontent.com/86747/218539569-decef497-ad10-44fd-91c8-9b0b311fbd8d.png)

@otavio Can you please help here?

**Comment by felipejoribeiro** [comment]
@gustavosbarreto , tentei o comando que mandou no último post.
Resultado segue:

![image](https://user-images.githubusercontent.com/39244876/218608435-e745da9e-0712-4f8d-a324-b5a09a9fa334.png)

gnu rules, btw

**Comment by otavio** [comment]
We need to understand what is causing your device not to work. Here:
```console
% docker pull --platform linux/arm/v7 shellhubio/agent:v0.9.4
v0.9.4: Pulling from shellhubio/agent
ac4f08a50f8c: Pull complete
ccfadbfba020: Pull complete
6b874748d339: Pull complete
58a82240d886: Pull complete
de57960c1451: Pull complete
1bb1cfe71b89: Pull complete
918c55bc5003: Pull complete
Digest: sha256:2e010e00897a9fd4a520a73b45031f88e16739935417d4c7471e8a15c8eb5042
Status: Downloaded newer image for shellhubio/agent:v0.9.4
docker.io/shellhubio/agent:v0.9.4
```

Please paste the output of:
```
docker info
```

**Comment by gustavosbarreto** [comment]
> @gustavosbarreto , tentei o comando que mandou no último post. Resultado segue:
> 
> ![image](https://user-images.githubusercontent.com/39244876/218608435-e745da9e-0712-4f8d-a324-b5a09a9fa334.png)
> 
> gnu rules, btw

Pode testar o outro comando?

```
docker manifest inspect shellhubio/agent:v0.9.4
```

**Comment by felipejoribeiro** [comment]
segue output:
```
pi@rasp:~ $ docker manifest inspect shellhubio/agent:v0.9.4
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1784,
         "digest": "sha256:3e5d6f97864d9377a97ac7569afea50de1e1aec49b77056c7ef8f93a1700125e",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1784,
         "digest": "sha256:884f40b10b46ef27d3c9644dac9c43a08d2e4f575b5430aeb165848e76f67441",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v6l"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1784,
         "digest": "sha256:de8b293cb077781a8e1370ba0392e8b05478543084cfb0d5b956a531c67fab4d",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v7l"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 1784,
         "digest": "sha256:0e029f36eadb51736ef3b18d9e7c771b2e667ab1396f66477eb2b237613873ec",
         "platform": {
            "architecture": "arm64",
            "os": "linux",
            "variant": "v8"
         }
      }
   ]
}
pi@rasp:~ $ docker pull --platform linux/arm/v7 shellhubio/agent:v0.9.4
v0.9.4: Pulling from shellhubio/agent
no matching manifest for linux/arm/v7 in the manifest list entries
```

**Comment by otavio** [comment]
The difference is the Docker Engine version:
- @otavio case: 20.10.23
- @felipejoribeiro case: 23.0.1

@felipejoribeiro, could you downgrade your engine version and see if it helps?

**Comment by gustavosbarreto** [comment]
> The difference is the Docker Engine version:
> 
> * @otavio case: 20.10.23
> * @felipejoribeiro case: 23.0.1
> 
> @felipejoribeiro, could you downgrade your engine version and see if it helps?

@otavio I'm using the same version as @felipejoribeiro and it works.

**Comment by otavio** [comment]
@gustavosbarreto you are using Arch Linux, right? @felipejoribeiro in your case this is Raspian and Docker repo?

**Comment by felipejoribeiro** [comment]
yes, we are using Raspian:
```
pi@rasp:~ $ cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
```

**Comment by felipejoribeiro** [comment]
The way we install docker in our systems is like so:
```
  curl -fsSL https://get.docker.com -o get-docker.sh &> /dev/null
  sudo sh get-docker.sh &> /dev/null
  sudo usermod -aG docker pi &> /dev/null
  sudo usermod -aG docker root &> /dev/null
  sudo systemctl enable docker &> /dev/null
  sudo systemctl start docker &> /dev/null
```

**Comment by gustavosbarreto** [comment]
@felipejoribeiro I found the issue and made a temporary fix for `v0.9.4` tag. You can proceed installing this old version.

This will be fixed in the next stable release.

**Comment by felipejoribeiro** [comment]
@gustavosbarreto Its working as expected now. Thank you very much! 🍻
